src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -410,6 +410,21 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
                             var boundAttribute = boundAttributeArray[i];
                             Debug.Assert(boundAttribute is not null);
                             NullableWalker.AnalyzeIfNeeded(binders[i], boundAttribute, boundAttribute.Syntax, diagnostics.DiagnosticBag);
+
+                            // Store attribute arguments for local usage analysis.
+                            if (!boundAttribute.ConstructorArguments.IsEmpty || !boundAttribute.NamedArguments.IsEmpty)
+                            {
+                                var localsExtractor = new LocalsExtractor();
+                                foreach (var argument in boundAttribute.ConstructorArguments)
+                                {
+                                    localsExtractor.Visit(argument);
+                                }
+                                foreach (var assignment in boundAttribute.NamedArguments)
+                                {
+                                    localsExtractor.Visit(assignment.Right);
+                                }
+                                boundAttributes[i].ReferencedLocals = localsExtractor.GetLocals();
```

Comment:
>boundAttributes[i].ReferencedLocals = localsExtractor.GetLocals(); [](http://example.com/codeflow?start=32&length=66)

Mutating the state of ```boundAttributes``` after ```SetAttributes``` call is probably not the right thing to do. Could introduce a "race" between different threads accessing the data.